From 824a8fd02f3c217255361d485d63830c32b82cf0 Mon Sep 17 00:00:00 2001
From: Jake <jake@fulu.org>
Date: Thu, 9 Jul 2026 01:08:16 +0100
Subject: [PATCH] Fix deprecation warning for null $birthday in
 formatBirthday()

up_birthday is nullable and is null for any user who hasn't set a
birthday (e.g. a freshly created account). Passing that straight to
explode() throws a PHP 8.1+ deprecation notice:
"explode(): Passing null to parameter #2 ($string) of type string is
deprecated".

---
 UserProfile/includes/specials/SpecialUpdateProfile.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/UserProfile/includes/specials/SpecialUpdateProfile.php b/UserProfile/includes/specials/SpecialUpdateProfile.php
index c7052f8c..d926aa61 100644
--- a/UserProfile/includes/specials/SpecialUpdateProfile.php
+++ b/UserProfile/includes/specials/SpecialUpdateProfile.php
@@ -337,7 +337,7 @@ class SpecialUpdateProfile extends UnlistedSpecialPage {
 	}
 
 	public static function formatBirthday( $birthday, $showYOB = false ) {
-		$dob = explode( '-', $birthday );
+		$dob = explode( '-', (string)$birthday );
 		if ( count( $dob ) == 3 ) {
 			$month = $dob[1];
 			$day = $dob[2];
-- 
2.43.0